### DISCUSSION TITLE
Google searchbox

### DISCUSSION THREAD

**Discussion by pvilas** [squidfunk/mkdocs-material#2295] [General]
Hi,

Does anybody know if it is possible to use and how we can configure the search box with google? In particular, what url should we write on the itemprop content:
```
<meta itemprop="target" content="https://query.example.com/search?q={search_term_string}"/>
```
 [documentation](https://developers.google.com/search/docs/data-types/sitelinks-searchbox)

**Comment by squidfunk** [comment]
If I understand correctly, you're asking for deep linking of search results. This is already possible with Insiders and the [`search.share`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#search-sharing) feature. With that enabled, you can do the following:

https://squidfunk.github.io/mkdocs-material-insiders/?q=change+color

**Reply by ssbarnea** [reply]
If I understood correctly, the OP asked about being able to use google search instead of alongside the built-in search. I am VERY interested about this and in fact that is what currently prevents us at Ansible to expand the use of mkdocs.

We have lots of projects under Ansible umbrella, each of them with its own docs built (sphinx or mkdocs) and we want to allow users to also make searches across the entire ecosystem,... and google search is by far the most-widely solution for this.

Basically we kinda need an option to hook an external search engine into the current build-in search option from mkdocs-material.

It does not have to replace the built-in as I still see a lot of value to be able to limit the search for current project if you want. I guess a "[x] global search" option the UI element to change the search behaviour?

**Reply by squidfunk** [reply]
That's definitely possible – some users are using Algolia with Material for MkDocs, so Google should be perfectly possible as well. For example, check https://docs.coregames.com/, a heavily customized version with Algolia integration.
